### PR TITLE
Hassio snapshot: download + design tweaks

### PR DESCRIPTION
**Pull request by c727**
- changed snapshot dialog to the new toolbar design
- snapshot size rounding with 1 decimal place
- hide `Folders` in details if empty
- snapshot delete button is now a icon-button
- snapshot download as icon-butto
- renamed restore buttons for better understanding
- add-ons in details are scrollable now
- changed some wording
- some code cleanup

ss

### CODE REVIEW

**Comment by c727** [comment]
screenshot fail ;D - the add-on list should be scrollable

could need help with the snapshot upload using the API. https://caniuse.com/#feat=filereader

File: hassio/snapshots/hassio-snapshot.html [hunk 1]
```diff
@@ -139,7 +177,7 @@ <h2>[[computeName(snapshot)]]</h2>
     return type === 'full';
   }
 
-  partialRestoreTapped() {
+  partialRestoreClicked() {
```
Comment (balloob) [inline hassio/snapshots/hassio-snapshot.html]: Prefix functions with `_` as a convention to show that they are private and only used inside your element.

File: hassio/snapshots/hassio-snapshot.html [hunk 2]
```diff
@@ -189,6 +227,19 @@ <h2>[[computeName(snapshot)]]</h2>
       });
   }
 
+  downloadClicked() {
+    this.hass.callApi('get', `hassio/snapshots/${this.snapshotSlug}/download`)
+      .then((data) => {
+        const element = document.createElement('a');
+        element.setAttribute('href', 'data:application/tar;charset=utf-8,' + encodeURIComponent(data));
```
[the remark this answers is not in the document]
Comment (c727) [reply hassio/snapshots/hassio-snapshot.html]: This doesn't work if api_password is used

[the remark this answers is not in the document]
Comment (c727) [reply hassio/snapshots/hassio-snapshot.html]: Hmz, I think I can read it from a cookie or session...

**Comment by c727** [comment]
snapshot download options:
1. `hass.callApi` -> memory -> HDD <-- "really really bad"
2. I save the api_password on login in a variable <-- I don't think that we want the keep the password stored plain in frontend forever
3. Request the password from the Hass.io backend

**Comment by c727** [comment]
should be OK now

**Review by balloob — APPROVED** [review]

**Pull request merged**